buildenv/jenkins/common/variables-functions.groovy

```diff
@@ -1361,6 +1366,26 @@ def set_build_extra_options(build_specs=null) {
         if (!EXTRA_MAKE_OPTIONS) {
             EXTRA_MAKE_OPTIONS = buildspec.getVectorField("extra_make_options", SDK_VERSION).join(" ")
         }
+        if (params.CODE_COVERAGE) {
+            // avoid children pipeline set CODE_COVERAGE option multiple times
+            if (!EXTRA_MAKE_OPTIONS.contains("-DCODE_COVERAGE=ON")) {
+                // add or append CODE_COVERAGE=ON option to EXTRA_CMAKE_ARGS
+                if (EXTRA_MAKE_OPTIONS.contains("EXTRA_CMAKE_ARGS=\"")) {
+                    EXTRA_MAKE_OPTIONS = EXTRA_MAKE_OPTIONS.replace("EXTRA_CMAKE_ARGS=\"", "EXTRA_CMAKE_ARGS=\"-DCODE_COVERAGE=ON ")
+                } else if (EXTRA_MAKE_OPTIONS.contains("EXTRA_CMAKE_ARGS=-D")) {
+                    sub_options = EXTRA_MAKE_OPTIONS.split(" ")
+                    sub_options.eachWithIndex { sub_option, index ->
+                        if (sub_option.contains("EXTRA_CMAKE_ARGS=")) {
+                            sub_option = sub_option.replace("EXTRA_CMAKE_ARGS=", "EXTRA_CMAKE_ARGS=\"-DCODE_COVERAGE=ON ")
+                            sub_options[index] = sub_option + "\""
+                        }
+                     }
+                    EXTRA_MAKE_OPTIONS = sub_options.join(" ")
+                } else {
+                    EXTRA_MAKE_OPTIONS += " EXTRA_CMAKE_ARGS=-DCODE_COVERAGE=ON"
+                }
```

Comment:
I don't think this handles all scenarios properly yet.
* `-DCODE_COVERAGE=ON` might be in `EXTRA_MAKE_OPTIONS`, but outside the definition of `EXTRA_CMAKE_ARGS`
* `-DCODE_COVERAGE=1` has the same meaning, but won't be recognized

I think the code should look for `EXTRA_MAKE_OPTIONS` and only alter that value (or add it if missing).

Reply:
This PR has been updated based on the suggestion.
Added handling cases to alter `-DCODE_COVERAGE=1`, tested working with
- `EXTRA_CMAKE_ARGS=-DCODE_COVERAGE=1`
- `EXTRA_CMAKE_ARGS="-DCODE_COVERAGE=1 -DXX=Y"`
- `EXTRA_CMAKE_ARGS="-DXX=Y -DCODE_COVERAGE=1"`

The other conditions are set to make sure only adding `-DCODE_COVERAGE=ON` after `EXTRA_CMAKE_ARGS`, to prevent the case that `-DCODE_COVERAGE=ON` in `EXTRA_MAKE_OPTIONS` directly . Thanks.

Reply:
Although not perfect, I think this is good enough for now given the limited access to defining `EXTRA_MAKE_OPTIONS` when launching a pipeline on Jenkins.